src/server/db_slice.cc

```diff
@@ -830,4 +852,28 @@ size_t DbSlice::EvictObjects(size_t memory_to_free, PrimeIterator it, DbTable* t
   return freed_memory_fun();
 };
 
+void DbSlice::RegisterWatchedKey(DbIndex db_indx, std::string_view key, ConnectionState::ExecInfo* exec_info) {
+  db_arr_[db_indx]->watched_keys[key].push_back(exec_info);
+}
+
+void DbSlice::UnregisterConnectionWatches(ConnectionState::ExecInfo* exec_info) {
+  for (const auto& [db_indx, key] : exec_info->watched_keys) {
+    auto& watched_keys = db_arr_[db_indx]->watched_keys;
+    if (auto it = watched_keys.find(key); it != watched_keys.end()) {
+      it->second.erase(std::remove(it->second.begin(), it->second.end(), exec_info),
+                       it->second.end());
+      if (it->second.empty())
+        watched_keys.erase(it);
+    }
+  }
+}
+
+void DbSlice::InvalidateDbWatches(DbIndex db_indx) {
+  for (const auto& [key, conn_list] : db_arr_[db_indx]->watched_keys) {
+    for (auto conn_ptr : conn_list) {
+      conn_ptr->watched_dirty.store(true);
```

Comment:
on aarch64 the default ordering is awfully slow. like could be several orders of magnitude slower.

Reply:
Ah, missed it